smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/EmitterImpl.java

```diff
@@ -7,29 +7,54 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import io.reactivex.BackpressureOverflowStrategy;
-import io.reactivex.BackpressureStrategy;
-import io.reactivex.Flowable;
-import io.reactivex.FlowableEmitter;
+import io.reactivex.*;
 import io.smallrye.reactive.messaging.annotations.Emitter;
+import io.smallrye.reactive.messaging.annotations.OnOverflow;
 
 public class EmitterImpl<T> implements Emitter<T> {
 
     private final AtomicReference<FlowableEmitter<Message<? extends T>>> internal = new AtomicReference<>();
     private final Flowable<Message<? extends T>> publisher;
-    private final String name;
 
     private static final Logger LOGGER = LoggerFactory.getLogger(EmitterImpl.class);
 
-    EmitterImpl(String name) {
-        this.name = name;
-        publisher = Flowable.<Message<? extends T>> create(fe -> {
+    EmitterImpl(String name, OnOverflow onOverflow) {
+        FlowableOnSubscribe<Message<? extends T>> deferred = fe -> {
             if (!internal.compareAndSet(null, fe)) {
                 fe.onError(new Exception("Emitter already created"));
             }
-        }, BackpressureStrategy.BUFFER)
-                .onBackpressureBuffer(128, () -> LOGGER.error("Buffer full for emitter {}", name),
-                        BackpressureOverflowStrategy.ERROR);
+        };
+        if (onOverflow == null) {
+            publisher = Flowable.create(deferred, BackpressureStrategy.BUFFER)
+                    .onBackpressureBuffer(128, () -> LOGGER.error("Buffer full for emitter {}", name),
+                            BackpressureOverflowStrategy.ERROR);
+        } else {
+            switch (onOverflow.value()) {
+                case BUFFER:
+                    Flowable<Message<? extends T>> p = Flowable.create(deferred, BackpressureStrategy.BUFFER);
+                    if (onOverflow.bufferSize() > 0) {
+                        publisher = p.onBackpressureBuffer(onOverflow.bufferSize(),
+                                () -> LOGGER.error("Buffer full for emitter {}", name), BackpressureOverflowStrategy.ERROR);
+                    } else {
+                        publisher = p;
```

Comment:
just a question here, is the buffer size set here to default value as described in the doc -127? It looks to me that the onBackpressureBuffer is not set at all if the bufferSize in annotation is not set which the defaults to 0

https://github.com/smallrye/smallrye-reactive-messaging/pull/175/files#diff-d47d3adc7e916519aa6946ebf2957af0R67

or am I missing something?

Reply:
By default, if `OnOverflow` is not used, we use a buffer of 127 elements (current behavior). We may want to add a way to configure the `127`magic number.

If `OnOverflow`is used with the buffer strategy, it would:

* use a buffer of the size passed in the `bufferSize` attribute
* use an unbounded buffer is `bufferSize` is not set

So, we cannot use a _default_ value for these cases, as it depends on the stream.

Reply:
I was referring to this line

```
if (onOverflow.bufferSize() > 0) {
```
where I assume `onOverflow` is annotation and the `bufferSize` is optional with default value 0. So if someone creates

```
@Inject
	@Stream("hello")
	@OnOverflow(value = OnOverflow.Strategy.BUFFER)
	Emitter<String> emitter;
```

then the back pressure is not set at all, isn't it? because it goes into the else statement and `p` is only 

```
Flowable<Message<? extends T>> p = Flowable.create(deferred, BackpressureStrategy.BUFFER);
```

Reply:
In this case, the backpressure is handled using an unbounded buffer. I'm going to extend the documentation to clarify this case.